data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/stream/LogicalReplicationClient.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ */
+
+package org.opensearch.dataprepper.plugins.source.rds.stream;
+
+import org.opensearch.dataprepper.plugins.source.rds.schema.ConnectionManager;
+import org.postgresql.PGConnection;
+import org.postgresql.replication.LogSequenceNumber;
+import org.postgresql.replication.PGReplicationStream;
+import org.postgresql.replication.fluent.logical.ChainedLogicalStreamBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.ByteBuffer;
+import java.sql.Connection;
+
+public class LogicalReplicationClient implements ReplicationLogClient {
+
+    private static final Logger LOG = LoggerFactory.getLogger(LogicalReplicationClient.class);
+
+    private final ConnectionManager connectionManager;
+    private final String replicationSlotName;
+    private LogSequenceNumber startLsn;
+    private LogicalReplicationEventProcessor eventProcessor;
+
+    private volatile boolean disconnectRequested = false;
+
+    public LogicalReplicationClient(final ConnectionManager connectionManager,
+                                    final String replicationSlotName) {
+        this.connectionManager = connectionManager;
+        this.replicationSlotName = replicationSlotName;
+    }
+
+    @Override
+    public void connect() {
+        PGReplicationStream stream;
+        try (Connection conn = connectionManager.getConnection()) {
+            PGConnection pgConnection = conn.unwrap(PGConnection.class);
+
+            // Create a replication stream
+            ChainedLogicalStreamBuilder logicalStreamBuilder = pgConnection.getReplicationAPI()
+                    .replicationStream()
+                    .logical()
+                    .withSlotName(replicationSlotName)
+                    .withSlotOption("proto_version", "1")
+                    .withSlotOption("publication_names", "my_publication");
```

Comment:
1. Move the constant to static variable
2. Use the created publication names.

Reply:
Done.